### PR TITLE
Gunpoints now take half a second to activate, make gasp sounds, and briefly immobilize the shooter and target, other small balance changes

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This PR messes around with gunpoints a bit, with the purpose of making them more viable in certain scenarios without making them obnoxious. The biggest change is that gunpoints now require a 0.5 second do_after() where neither the shooter nor the target moves, and immobilizes both of them for 0.75 seconds if point blank, or half that if you're 2 tiles away. Originally you were supposed to only be able to initiate a gunpoint from point-blank, but #56601 seems to have removed that requirement, so we'll run with it and just leave it as advantageous to gunpoint closer up. The do_after() reinforces that it should be used as an ambush tactic, and so you can't use it on someone who's actively fleeing or fighting you.

Getting held up will now make you emit a shocked gasp sound, a la Metal Gear Solid, which combined with the short immobilize will hopefully make it more noticeable that someone's pointing a gun at you.

Holdups will now immediately give a 25% bonus to damage and wounds, instead of having to wait 2.5 seconds to hit the double damage stage.

Finally, right clicking someone that you're holding up will no longer shoot them. That just feels like good consistency.

## Why It's Good For The Game
Hopefully makes gunpoints a little more viable for when you want to stick someone who's not expecting it up without them immediately jetting off. In the future I'd like to ape Baycode and let the gunman have an action that toggles whether the victim is allowed to move, so you can order them to move to a second location without instantly shooting them, but that'll come later.
## Changelog
:cl: Ryll/Shaps
balance: Holding someone at gunpoint now requires both the shooter and the victim to hold still for half a second before activating, so you can't hold-up people fleeing or fighting you. After that, it will briefly immobilize the both of you, 0.75 seconds if adjacent, or half that if you're two tiles away. Nuke ops are immune to the immobilization, since they're ready to die anyways.
balance: Holding someone up will immediately apply a 1.25x damage and wound multiplier, rather than waiting 2.5 seconds to hit 2x.
soundadd: Being held up will now make the victim play a sharp gasp sound, a la Metal Gear Solid.
qol: Trying to hold someone up that you're already holding up will no longer shoot them.
/:cl:

### CODE REVIEW

**Comment by TheBoondock** [comment]
Another based Ryll pr

**Comment by Livrah** [comment]
I already see a problem on warops rounds.

**Comment by Ryll-Ryll** [comment]
> I already see a problem on warops rounds. Why is this instant.

Please use your words to clearly state and lay out your concerns instead of this

**Comment by Cheshify** [comment]
> I already see a problem on warops rounds. Why is this instant.

If there's a nukie within 2 tiles of you isn't than an issue for either party?

**Comment by Livrah** [comment]
> If there's a nukie within 2 tiles of you isn't than an issue for either party?

It's problem with one vs station situations. I bet this thing ignores block chance, shields like nukie / wiz MODs have, etc. 
Every clown with empty gun now can stun you in 2 tiles range instantly, sounds very bad. Any delay would be good so you can't stun moving targets.

**Comment by Cheshify** [comment]
> > If there's a nukie within 2 tiles of you isn't than an issue for either party?
> 
> It's problem with one vs station situations. I bet this thing ignores block chance, shields like nukie / wiz MODs have, etc. Every clown with empty gun now can stun you in 2 tiles range instantly, sounds very bad. Any delay would be good so you can't stun moving targets.

That's a good point. Maybe a half-second delay or something?

**Comment by Livrah** [comment]
> 

You should apply this to all antagonists then. And yes it will be a problem with antagchecks.

**Comment by Livrah** [comment]
> > > If there's a nukie within 2 tiles of you isn't than an issue for either party?
> > 
> > 
> > It's problem with one vs station situations. I bet this thing ignores block chance, shields like nukie / wiz MODs have, etc. Every clown with empty gun now can stun you in 2 tiles range instantly, sounds very bad. Any delay would be good so you can't stun moving targets.
> 
> That's a good point. Maybe a half-second delay or something?

Makes much more sense. I'd like to see more pointblank as a surprise thing, but not combat abuse strats.

**Comment by Cheshify** [comment]
Don't have it apply to "all antags", nobody should get a resistance to it. It's fine if there's a short timer on it.

**Comment by Ryll-Ryll** [comment]
I'm just gonna apply the immunity to military types like operatives and maybe ERT's. MAYBE sleeping carp users in certain cases?

**Comment by itseasytosee** [comment]
I don't like how stamina damage is being used here. Feels really arbitrary. Why not just have a cooldown?

**Comment by Dezupher** [comment]
I think the best way to handle it (in addition to blanket giving stuff like ops immunity to the hold up) would be being on combat mode made you immune to the immobilization.

If I get held up by someone as I'm running, and then their buddy catches up to me and stun batons me thanks to the immobilize, I'm gonna be upset.

**Comment by necromanceranne** [comment]
> I don't like how stamina damage is being used here. Feels really arbitrary. Why not just have a cooldown?

I think arbitrary self harm over simply an internal cooldown is definitely the wrong way to handle it, so I agree with this assessment as well.

I also think combat mode making you immune to the immobilization (as well as the gunman not being affected in this case) is...hard to see as a usable solution to that problem, since combat mode isn't a visible status to other people, and combat mode is unfortunately not a consistently utilized system, so sometimes people sit on combat mode unintentionally because they happened to be, so, butchering meat or interacting with some machine that needed combat mode to do a specific interaction. (Sort of like how people used to sit on harm intent while we still had intents). It's maybe better than the current implementation, but I can't say for sure if it's a perfect solution.

**Comment by Comxy** [comment]
Maybe have it so there is a do_after a second or something of the shooter raising their gun with a balloon popup for nearby players, I think that would solve the antag checks.

**Comment by blackdav123** [comment]
the one time I tried to hold up a captain he immediately moved so neither of us had any chance to talk good change

**Comment by moocowswag** [comment]
Make it so you dont get immobilized if youre in combat mode, it shouldnt be a shock to you that a gun is being held up to you if youre already looking to fight.

**Comment by optimumtact** [comment]
the problem with that is there's no visual indication to other people that you're in combat mode which means the feature just appears to be broken.

There's no easy fixes to a two way signalling problem.

**Comment by AlamoTurtle** [comment]
Maybe give people who were recently held up an "on-edge" type of buff that lets them ignore the stun for some time, perhaps for a minute after successfully or unsuccessfully leaving a hold-up? The indicator for this potentially being something like the jittering effect (like people suffering from withdrawal) and the (somewhat) neutral moodlet of "i'm nervous and on the edge" stuff. This can potentially lead to anyone who's jittery and on edge who "can't stand still" being much more capable of reacting to a hold-up.

**Comment by ATH1909** [comment]
> Holdups will now immediately give a 25% bonus to damage and wounds, instead of having to wait 2.5 seconds to hit the double damage stage. The shooter suffers 20 stamina damage when initiating the holdup though, to prevent people from spamming it in normal combat.

some people are just hard as nails

**Comment by ATH1909** [comment]
can hulks hold people at gunpoint? if so, can they (or someone with stamina regen chems) spam this ability to keep someone immobilized indefinitely, provided they have an easy way to clear the gunpoint status (putting away the gun?)?

**Comment by Thelolswat** [comment]
cowboy stickup stealthops getting the disk and getting out before captain can tell sec that he was robbed by the van der linde gang

**Comment by Vect0r2** [comment]
What if it works in reverse? It doesn’t shoot for like 0.5-1 sec?

**Comment by Ryll-Ryll** [comment]
~ ~ There is now a 0.5 second do_after before the holdup applies, so if either the shooter or the target moves or gets interrupted, it won't apply ~ ~

In addition, the stamina damage has been removed, and nuke ops are now immune to the immobilize. This should prevent people from abusing gunpoints to hold up a fleeing enemy, while still allowing them to be used for ambushes.

**Comment by Striders13** [comment]
nuke ops specifically having immunity is so bizarre. Why not wizard? Why not any other solo antag?

**Comment by necromanceranne** [comment]
> nuke ops specifically having immunity is so bizarre. Why not wizard? Why not any other solo antag?

I can answer that pretty easily. It's called 'antag checking'.

File: code/datums/components/gunpoint.dm [hunk 1]
```diff
@@ -33,26 +37,33 @@
...
 	RegisterSignals(weapon, list(COMSIG_ITEM_DROPPED, COMSIG_ITEM_EQUIPPED), PROC_REF(cancel))
 
-	shooter.visible_message(span_danger("[shooter] aims [weapon] point blank at [target]!"), \
-		span_danger("You aim [weapon] point blank at [target]!"), ignored_mobs = target)
-	to_chat(target, span_userdanger("[shooter] aims [weapon] point blank at you!"))
-	add_memory_in_range(target, 7, /datum/memory/held_at_gunpoint, protagonist = target, deuteragonist = shooter, antagonist = weapon)
+	var/distance = min(get_dist(shooter, target), 1) // treat 0 distance as adjacent
+	var/distance_description = (distance <= 1 ? "point blank " : "")
+
+	shooter.visible_message(span_danger("[shooter] aims [weapon] [distance_description]at [target]!"), \
```
Comment (san7890) [inline code/datums/components/gunpoint.dm:47]:
```suggestion
	shooter.visible_message(span_danger("[shooter] aims [weapon] [distance_description]at [target]!"),
```

we don't need this for procs

**Comment by Jacquerel** [comment]
This PR seems to have stalled. I think changing gunpoints is cool but am not sure about what the effect of this would ultimately be. Worth some testmerging to see what players do with it?

**Comment by SinfulBliss** [comment]
This is quite strong, perhaps it's intended but seccies can fill their shotguns with improv and then rightclick you while you're fleeing point blank to get a free +25% damage and wound bonus (which, with improv is very strong). 

For instance if you're running from an officer, facing away, it's a bit weird that your character would get held up and then blasted for moving, even if they're point blank.

**Comment by Overuseddart** [comment]
Allows for a stun with a cap gun, or a permastun with 2 people holding at gunpoint. Just back them into a corner with combat and you can start stunning.

**Comment by Ryll-Ryll** [comment]
> Allows for a stun with a cap gun, or a permastun with 2 people holding at gunpoint. Just back them into a corner with combat and you can start stunning.

cap gun is not a real gun

**Comment by Higgin** [comment]
Would love to see this get a TM and how it feels in practice.

**Comment by Grmmis** [comment]
> Would love to see this get a TM and how it feels in practice.

Same here, I like using gunpoints albeit their bugginess.  Think this could make them work a lot better instead of just someone moving before they notice and getting shot.  TM would be cool: ).

**Comment by anclandor** [comment]
The wind up seems like it makes holding people up kinda useless.  I could see the shooter having a windup of not being able to move, but for the target to be able to just keep moving and avoid it seems a little extreme.

**Comment by optimumtact** [comment]
I'm not really sure where this is going but I trust you enough to merge it if it's ready

**Pull request merged**